### ISSUE TITLE
Error Compiling

### ISSUE THREAD

**Issue by RInRinKK** [progmem/Switch-Fightstick#10]
Hi,
I don't know if you work on this anymore and I'm sorry if this is a stupid question but, when I try to compile your code, I always get this:

`In file included from Descriptors.c:1:
Descriptors.h:50: error: conflicting types for 'CALLBACK_USB_GetDescriptor'
C:/SwitchFightstick/LUFA/LUFA/../LUFA/Drivers/USB/Core/AVR8/../Device.h:133: error: previous declaration of 'CALLBACK_USB_GetDescriptor' was here
Descriptors.c:155: error: conflicting types for 'CALLBACK_USB_GetDescriptor'
C:/SwitchFightstick/LUFA/LUFA/../LUFA/Drivers/USB/Core/AVR8/../Device.h:133: error: previous declaration of 'CALLBACK_USB_GetDescriptor' was here
make: *** [obj/Descriptors.o] Error 1`

And also

`Joystick.c: In function 'SetupHardware':
Joystick.c:120: warning: implicit declaration of function 'clock_prescale_set'
Joystick.c:120: error: 'clock_div_1' undeclared (first use in this function)
Joystick.c:120: error: (Each undeclared identifier is reported only once
Joystick.c:120: error: for each function it appears in.)
make: *** [obj/Joystick.o] Error 1`

I don't know much about this but it would be great if you could help me.

**Comment by progmem** [comment]
For the first error, I want to say someone pointed this out before and I haven't had a chance to change it. The newest version of LUFA has a different declaration of `CALLBACK_USB_GetDescriptor` that uses a `uint16_t` for the `wIndex` argument. This is a pretty simple fix: change the `wIndex` argument in Descriptors.c and Descriptors.h to use `uint16_t`.

For the second issue, `clock_prescale_set` and `clock_div_1` are components in avr-libc, in avr/power.h . Just to make sure, do you have WinAVR installed or some form of avr-libc available?

**Comment by RInRinKK** [comment]
Sorry for the late response, but I only had a chance to try it now.
For the first error I was able to solve it and for the second I compiled it via Linux and it went well.
I just have some questions about the initial setup,
When I finished flashing the hex file it appears as a POKKEN CONTROLLER as expected but is always pressing the button 8 on my PC. Also my Switch wont recognize it.
What should I do??

**Comment by progmem** [comment]
Are you flashing to a development board? If so, which one? (Teensy2, etc.) There's a possibility that the pin that represents pin 8 is being grounded by something else.

As for the Switch not recognizing it, I would have to recompile the code with the latest LUFA and flash to one of my AVR devboards for testing. I haven't made any changes with the descriptors for a year or so.

**Comment by RInRinKK** [comment]
I'm actually trying to flash to an arduino (it's a UNO R3, I'm using it in the atmega16u2).
Maybe you can tell me which LUFA library you used and the date so I can try to find it on the internet??

**Comment by progmem** [comment]
That explains why button 8 is pressed; PD7 is held low with a 1k pulldown. Keep in mind that using the 16u2 on an Arduino is going to be limited pin-wise; only 9 of the pins have direct exposure.

Has the target chip in the makefile been updated for the 16u2?

**Comment by RInRinKK** [comment]
Yes, I changed the makefile according to my needs.
 Anything I can do about PD7?
Also I think I have almost all of the pins actually
I've found this diagram:
![Arduino-UNO-pinout](https://user-images.githubusercontent.com/45297012/56616946-8e0c5f80-6616-11e9-8dfa-3b6ea083d45d.jpg)
What do you think?

**Comment by progmem** [comment]
You could alter the firmware to not read from that pin in the function, or you could try to read from a different pin instead for pin 8. By default it's reading all of ports D and B, but there's no reason why you can't customize this.

Also, I'm trying to find a schematic that matches what's being shown and I'm coming up dry. The Uno SMD schematic and the Uno R3 schematic that I'm finding only show 9 pins mapped to the 16u2 that could be used as data. Two others could be used, but those power the LEDs for serial send/receive.

**Comment by progmem** [comment]
Regarding the Switch detecting the controller, I recall this problem and it's a simple fix: a button press needs to be actuated first for the Switch to detect it. It has to be a press that occurs after the controller is plugged in, so the button that's being held down won't trigger it.

**Comment by progmem** [comment]
I'm going to go ahead and close this issue for now, as no other questions/concerns were brought up. Hope everything is working OK!

**Comment by fz3ateam** [comment]
im facing same issue here. i downloaded avr-libc but still shows same error

 [GCC]     : Compiling C file "Joystick.c"
avr-gcc -c -pipe -gdwarf-2 -g2 -mmcu=atmega16u2 -fshort-enums -fno-inline-small-functions -Wall -fno-strict-aliasing -funsigned-char -funsigned-bitfields -ffunction-sections -I. -DARCH=ARCH_AVR8 -DF_CPU=16000000UL -mrelax -fno-jump-tables -x c -Os -std=gnu99 -Wstrict-prototypes -DUSE_LUFA_CONFIG_HEADER -IConfig/   -I. -I./lufa/LUFA/.. -DARCH=ARCH_AVR8 -DBOARD=BOARD_NONE -DF_USB=16000000UL   -MMD -MP -MF obj/Joystick.d Joystick.c -o obj/Joystick.o
Assembler messages:
Fatal error: can't create obj/Joystick.o: No such file or directory
Joystick.c: In function 'SetupHardware':
Joystick.c:120: warning: implicit declaration of function 'clock_prescale_set'
Joystick.c:120: error: 'clock_div_1' undeclared (first use in this function)
Joystick.c:120: error: (Each undeclared identifier is reported only once
Joystick.c:120: error: for each function it appears in.)

**Comment by progmem** [comment]
@fz3ateam See [this comment](https://github.com/progmem/Switch-Fightstick/issues/14#issuecomment-591423207). This appears to be an issue with certain `avr-libc` versions, especially within WinAVR.

**Comment by fz3ateam** [comment]
after updating avr8 win32 in my windows 10.. I'm facing new issue


[GCC] : Compiling C file "Joystick.c"
avr-gcc -c -pipe -gdwarf-2 -g2 -mmcu=atmega16u2 -fshort-enums -fno-inline-small-functions -Wall -fno-strict-aliasing -funsigned-char -funsigned-bitfields -ffunction-sections -I. -DARCH=ARCH_AVR8 -DF_CPU=16000000UL -mrelax -fno-jump-tables -x c -Os -std=gnu99 -Wstrict-prototypes -DUSE_LUFA_CONFIG_HEADER -IConfig/ -I. -I./lufa/LUFA/.. -DARCH=ARCH_AVR8 -DBOARD=BOARD_NONE -DF_USB=16000000UL -MMD -MP -MF obj/Joystick.d Joystick.c -o obj/Joystick.o
Assembler messages:
Fatal error: can't create obj/Joystick.o: No such file or directory
Joystick.c:235:1: fatal error: error writing to -: Invalid argument
}
^
compilation terminated.
make: *** [obj/Joystick.o] Error 1
